dandi/upload.py

```diff
@@ -422,9 +422,8 @@ def ensure_folder():
                 item_rec["_id"], path.name, path.absolute()
             )
             if not current:
-                raise RuntimeError(
-                    "Must not happen since file %s was just uploaded" % path
-                )
+                yield skip_file("File size on server does not match local file size")
```

Comment:
I think the message is not accurate/too specific
```suggestion
                yield skip_file("File on server was unexpectedly changed")
```